### ISSUE TITLE
Architecture refactoring

### ISSUE THREAD

**Issue by alenmagdic** [aconno/BleSniffer#13]
Clean architecture has not been implemented properly, so some activities communicate directly with use case objects instead of through view models. View models should be created for each activity and activities should be changed in order to use this view models. Also, Dagger modules should be refactored so that activity modules provide only view models and so that there are separate modules for specific types of objects (for example, there should be DeserializerModule that provides use cases that are linked to deserializers).

**Comment by borna-mdn** [comment]
I agree, but I don't want to have an infinite refactoring task that will result in huge PR.

Propose a concrete refactoring task with:
* code samples - part of the code that is problematic / you want to refactor
* solution - changes you want to make
* benefits - improved performance, increased code coverage...

Also, we can create multiple sub-issues from this issue.
